src/Morphir/JsonSchema/Backend.elm

```
@@ -105,55 +105,144 @@ extractTypes modName definition =
-
-                "Morphir.SDK:Basics:float" ->
-                    Just Number
-
-                "Morphir.SDK:Basics:bool" ->
-                    Just Boolean
-
-                "Morphir.SDK:List:list" ->
-                    List.head types
-                        |> Maybe.andThen mapType
-                        |> Maybe.map Array
+        Type.Reference _ (( packageName, moduleName, localName ) as fQName) argTypes ->
+            case ( FQName.toString fQName, argTypes ) of
+                ( "Morphir.SDK:Basics:int", [] ) ->
+                    Ok Integer
+
+                ( "Morphir.SDK:Decimal:decimal", [] ) ->
+                    Ok String
+
+                ( "Morphir.SDK:String:string", [] ) ->
+                    Ok String
+
+                ( "Morphir.SDK:Char:char", [] ) ->
+                    Ok String
+
+                ( "Morphir.SDK:Basics:float", [] ) ->
+                    Ok Number
+
+                ( "Morphir.SDK:Tuple:tuple", [] ) ->
+                    Ok Number
+
+                ( "Morphir.SDK:Basics:bool", [] ) ->
+                    Ok Boolean
+
+                ( "Morphir.SDK:List:list", [ itemType ] ) ->
+                    Result.map2 Array
+                        (mapType itemType
+                            |> Result.map ListType
+                        )
+                        (Ok False)
+
+                ( "Morphir.SDK:Set:set", [ itemType ] ) ->
+                    Result.map2 Array
+                        (mapType itemType
+                            |> Result.map ListType
+                        )
+                        (Ok True)
+
+                ( "Morphir.SDK:Maybe:maybe", [ itemType ] ) ->
+                    Ok
+                        (OneOf
+                            [ Null
+                            , case mapType itemType of
+                                Ok val ->
+                                    val
+
+                                _ ->
+                                    Null
```

Same here. You should propagate the result instead of ignoring the error.

---

Fixed now